internal/endtoend/endtoend_test.go

```diff
@@ -95,52 +110,103 @@ func TestReplay(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	contexts := map[string]textContext{
+		"base": {
+			Mutate:  func(c *config.Config) {},
+			Enabled: func() bool { return true },
+		},
+		"managed-db": {
+			Mutate: func(c *config.Config) {
+				c.Cloud.Project = "01HAQMMECEYQYKFJN8MP16QC41" // TODO: Read from environment
+				for i := range c.SQL {
+					c.SQL[i].Database = &config.Database{
+						Managed: true,
+					}
+				}
+			},
+			Enabled: func() bool {
+				if len(os.Getenv("CI")) > 0 {
+					return false
+				}
+				return len(os.Getenv("SQLC_AUTH_TOKEN")) > 0
+			},
+		},
+	}
+
 	for _, replay := range dirs {
 		tc := replay
-		t.Run(tc, func(t *testing.T) {
-			t.Parallel()
+		for name, testctx := range contexts {
```

Comment:
I think this might be simpler if we set the default contexts explicitly in `parseExec` when none are given, like we do with command. Then we could move this for loop inside `t.Run` and just iterate over the contexts we have in `args.Contexts`, and we wouldn't need to `Skipf` any contexts.

But I guess if you explicitly want an independent test for each context then that doesn't work.